uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/BAO/Assignment.php

```diff
@@ -21,4 +21,91 @@ public static function create($params) {
 
     return $instance;
   } */
+
+
+  /**
+   * Get all cases with no end dates
+   *
+   * @param array $params
+   * @param array $excludeCaseIds
+   * @param bool $excludeDeleted
+   *
+   * @return array of case and related data keyed on case id
+   */
+  public static function getUnclosedCases($params = array(), $excludeCaseIds = array(), $excludeDeleted = TRUE, $includeClosed = FALSE) {
+    //params from ajax call.
+    $where = array($includeClosed ? '(1)' : '(ca.end_date is null)');
+    if ($caseType = CRM_Utils_Array::value('case_type', $params)) {
+      $where[] = "( civicrm_case_type.title LIKE '%$caseType%' )";
+    }
+    if ($sortName = CRM_Utils_Array::value('sort_name', $params)) {
+      $config = CRM_Core_Config::singleton();
+      $search = ($config->includeWildCardInName) ? "%$sortName%" : "$sortName%";
+      $where[] = "( sort_name LIKE '$search' )";
+    }
+    if ($cid = CRM_Utils_Array::value('contact_id', $params)) {
+      $where[] = " c.id = $cid ";
+    }
+    if (is_array($excludeCaseIds) &&
+      !CRM_Utils_System::isNull($excludeCaseIds)
+    ) {
+      $where[] = ' ( ca.id NOT IN ( ' . implode(',', $excludeCaseIds) . ' ) ) ';
+    }
+    if ($excludeDeleted) {
+      $where[] = ' ( ca.is_deleted = 0 OR ca.is_deleted IS NULL ) ';
+    }
+
+    //filter for permissioned cases.
+    $filterCases = array();
+    $doFilterCases = FALSE;
+    if (!CRM_Core_Permission::check('access all cases and activities')) {
+      $doFilterCases = TRUE;
+      $session = CRM_Core_Session::singleton();
+      $filterCases = CRM_Case_BAO_Case::getCases(FALSE, $session->get('userID'));
+    }
+    $whereClause = implode(' AND ', $where);
+
+    $limitClause = '';
+    if ($limit = CRM_Utils_Array::value('limit', $params)) {
+      $limitClause = "LIMIT 0, $limit";
+    }
+
+    $query = "
+    SELECT  c.id as contact_id,
+            c.sort_name,
+            ca.id,
+            ca.subject as case_subject,
+            civicrm_case_type.title as case_type,
+            ca.start_date as start_date,
+            ca.end_date as end_date,
+            ca.status_id
+      FROM  civicrm_case ca INNER JOIN civicrm_case_contact cc ON ca.id=cc.case_id
+ INNER JOIN  civicrm_contact c ON cc.contact_id=c.id
+ INNER JOIN  civicrm_case_type ON ca.case_type_id = civicrm_case_type.id
+     WHERE  {$whereClause}
+  ORDER BY  c.sort_name, ca.end_date
+            {$limitClause}
+";
```

Comment:
hmmmmm... I was thinking that maybe you could use the JOIN options on the API to fetch the data. Something like this:
![captura de tela 2016-08-30 as 07 43 30](https://cloud.githubusercontent.com/assets/388373/18086289/905b73f6-6e85-11e6-8a18-c696f43864f7.png)

but I just tried it here and it's giving me an error (maybe a bug in civicrm or me doing something wrong):
![captura de tela 2016-08-30 as 07 45 09](https://cloud.githubusercontent.com/assets/388373/18086321/b67991c6-6e85-11e6-8823-3516f769cb51.png)

Anyway, if it's really not possible to do it using the API or if it will hurt performance too much, then I think it's fine to keep using a SQL query